### ISSUE TITLE
Input data specifications?

### ISSUE THREAD

**Issue by schroeme** [BayraktarLab/cell2fate#4]
Hi! Can you please specify what is required for input data (file format, structure, layers, annotations, etc.)? For example, in `cell2fate_PancreasWithCC.ipynb`, `adata = sc.read_h5ad(data_path + data_name + '_anndata.h5ad')` presumably reads in a counts matrix. Later, `c2f.Cell2fate_DynamicalModel.setup_anndata(adata, spliced_label='spliced', unspliced_label='unspliced')` presumably specifies layer names for the spliced and unspliced counts.

Are there two counts matrices in different layers of the AnnData object, one for spliced and one for unspliced counts for each gene? If so, can you recommend a tool for estimated spliced vs. unspliced counts? We use 10x Cell Ranger to align and count our reads, and I don't think counts are annotated as spliced or unspliced.

Apologies if I missed this in the preprint or somewhere in the documentation.

**Comment by AlexanderAivazidis** [comment]
Hi! Thanks for pointing out this missing information in the tutorials. I will update the tutorials!

For now let me explain that we do indeed expect adata.layers['spliced'] to contain spliced raw umi counts and adata.layers['unspliced'] to contain the unspliced raw umi counts.

To obtain those for our new developing human brain single-nucleus RNAseq data we used starsolo to do count quantification: https://github.com/alexdobin/STAR/blob/master/docs/STARsolo.md

It has the option:

--soloFeatures Gene Velocyto

To also quantify spliced and unspliced counts similar to how it is done in the velocyto method (http://velocyto.org/). You can also just run velocyto in addition to running cell ranger. I believe velocyto is still the most widely used spliced/unspliced count quantification method. and was used for all the data in our paper. Also this recent developing human brain atlas used cellranger+velocyto :

https://github.com/linnarsson-lab/developing-human-brain

There are however also other spliced/unspliced quantification methods available and this paper gives a good overview it seems: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7822509/pdf/pcbi.1008585.pdf

**Comment by schroeme** [comment]
This is really helpful, thanks! I'll run velocyto - seems I can do this directly from Cell Ranger output files.

**Comment by AlexanderAivazidis** [comment]
Ok great! I have now also added this information to the tutorials. Thanks for trying the method at such an early stage! Please get in touch if you have any more issues.

**Comment by SevenTea7** [comment]
> 这真的很有帮助，谢谢！我可以直接从Cell Ranger输出文件中运行velocyto。

Hello, I want to know if I got loom from velocyto. Is this correct? When I obtained h5ad for this model following scvelo's method, I reported an error at "mod.train()", "ValueError:" Expected value argument (Tensor of shape (1000, 3000,  2)) to be within the support (IntegerGreaterThan(lower_bound=0)) of the distribution GammaPoisson(),  but found invalid values. May I ask if there are any solutions? Thank you
